core/arch/aarchxx/mangle.c

```diff
@@ -432,21 +554,18 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
         XINST_CREATE_add(dcontext, opnd_create_reg(DR_REG_X0),
                          OPND_CREATE_INT16(dstack_offs - 32 * XSP_SZ)));
 
-    /* Push SIMD registers. */
-    insert_save_registers(dcontext, ilist, instr, cci->simd_skip, DR_REG_X0, DR_REG_Q0,
-                          SIMD_REG_TYPE);
-
     if (proc_has_feature(FEATURE_SVE)) {
-        /* add x0, x0, #(dstack_offs) */
-        PRE(ilist, instr,
-            XINST_CREATE_add(dcontext, opnd_create_reg(DR_REG_X0),
-                             OPND_CREATE_INT16(512)));
+        /* Save the SVE regs */
+        insert_save_registers(dcontext, ilist, instr, cci->simd_skip, DR_REG_X0,
+                              DR_REG_Z0, SVE_ZREG_TYPE);
+    } else {
+        /* Save the SIMD registers. */
         insert_save_registers(dcontext, ilist, instr, cci->simd_skip, DR_REG_X0,
-                              DR_REG_Z0, SVE_REG_TYPE);
+                              DR_REG_Q0, SIMD_REG_TYPE);
     }
 
-    dstack_offs += (proc_num_simd_registers() * sizeof(dr_simd_t));
-    ASSERT(proc_num_simd_registers() == MCXT_NUM_SIMD_SLOTS);
+    dstack_offs += (MCXT_NUM_SIMD_SLOTS + MCXT_NUM_SVEP_SLOTS + MCXT_NUM_FFR_SLOTS) *
```

Comment:
Here's the same sum again: see the prior comment.  Do we need the max, or does the dynamic suffice?  If we need the max: how about a helper to avoid repeating this (I see it below as well).

Reply:
Replaced with `MCXT_NUM_SIMD_SVE_SLOTS` in 5 instances.